src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentProperty.java

```
@@ -154,51 +155,86 @@ private void initDateConverter() {
+								field.type().name()));
+					}
 
-				converter = ElasticsearchDateConverter.of(pattern);
-			} else {
-
-				switch (dateFormat) {
-					case weekyear:
-					case weekyear_week:
-					case weekyear_week_day:
-						LOGGER.warn("no Converter available for " + actualType.getName() + " and date format " + dateFormat.name()
-								+ ". Use a custom converter instead");
-						break;
-					default:
-						converter = ElasticsearchDateConverter.of(dateFormat);
-						break;
+					for (int j = 0; j < dateFormatPatterns.length; j++) {
+						String pattern = dateFormatPatterns[j];
+						if (!StringUtils.hasText(pattern)) {
+							throw new MappingException(
+									String.format("Date pattern of property %s must not be empty", property, field.type().name()));
+						}
+						converters.add(ElasticsearchDateConverter.of(pattern));
+					}
+				} else {
+
+					switch (dateFormat) {
+						case weekyear:
+						case weekyear_week:
+						case weekyear_week_day:
+							LOGGER.warn("No default converter available for " + actualType.getName() + " and date format "
+									+ dateFormat.name() + ". Use a custom converter instead.");
+							break;
+						default:
+							converters.add(ElasticsearchDateConverter.of(dateFormat));
+							break;
+					}
 				}
+
 			}
 
-			if (converter != null) {
-				ElasticsearchDateConverter finalConverter = converter;
+			if (!converters.isEmpty()) {
 				propertyConverter = new ElasticsearchPersistentPropertyConverter() {
-					final ElasticsearchDateConverter dateConverter = finalConverter;
+					final List<ElasticsearchDateConverter> dateConverters = converters;
+
+					@SuppressWarnings("unchecked")
+					@Override
+					public Object read(String s) {
+						if (isTemporalAccessor) {
+							for (ElasticsearchDateConverter dateConverter : dateConverters) {
+								try {
+									return dateConverter.parse(s, (Class<? extends TemporalAccessor>) actualType);
+								} catch (Exception e) {
+								}
+							}
```

after the loop, if every registered converter fails, we should throw an error that none of the defined date formats could be used to read the value